ddl/db_test.go

```diff
@@ -1470,3 +1474,120 @@ func TestReportingMinStartTimestamp(t *testing.T) {
 	infoSyncer.ReportMinStartTS(dom.Store())
 	require.Equal(t, validTS, infoSyncer.GetMinStartTS())
 }
+
+// for issue #34931
+func TestBuildMaxLengthIndexWithNonRestrictedSqlMode(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+
+	r := tk.MustQuery("select @@sql_mode")
+	defaultSqlMode := r.Rows()[0][0].(string)
```

Comment:
```suggestion
	defaultSQLMode := r.Rows()[0][0].(string)
```
From https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check/detail/tidb_ghpr_check/65639/pipeline:

```
[2022-06-01T12:24:31.444Z] linting
[2022-06-01T12:24:31.444Z]   ✘  https://revive.run/r#var-naming  var defaultSqlMode should be defaultSQLMode  
[2022-06-01T12:24:31.444Z]   ddl/db_test.go:1487:2
```

Reply:
fixed